src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/DebugDirectoryNode.cs

```diff
@@ -88,8 +89,30 @@ int GetNumDebugDirectoryEntriesInModule()
             if (_module == null)
                 return 0;
 
-            ImmutableArray<DebugDirectoryEntry> entries = _module.PEReader.ReadDebugDirectory();
-            return entries == null ? 0 : entries.Length;
+            PEReader moduleReader = _module.PEReader;
+            ImmutableArray<DebugDirectoryEntry> entries = ImmutableArray<DebugDirectoryEntry>.Empty;
+
+            try
+            {
+                entries = moduleReader.ReadDebugDirectory();
+            }
+            catch (BadImageFormatException e)
+            {
+                if (e.Message.Equals("Invalid directory size."))
+                {
+                    int actualDbgDirSize = moduleReader.PEHeaders.PEHeader.DebugTableDirectory.Size;
+
+                    // This comes from the Size property of the DebugDirectoryEntry class.
+                    int expectedDbgDirSizeBase = 28;
+
+                    Console.WriteLine("WARNING: This image's Debug Directory Entry size should be a"
```

Comment:
these can be created as exception strings instead of string literals?

Reply:
I agree we should make this an exception string. I would throw away the "artificial intelligence" analyzing the exception message and just print something generic like "WARNING: %s contains invalid debug information: %s" with the name of the file and the message of the exception caught.

Reply:
I think we can afford to keep the part that says "Debugging tools may not work correctly with this image". I feel our users would like to know what the warning entails for them.